js_modules/dagster-ui/packages/ui-core/src/assets/__tests__/LaunchAssetChoosePartitionsDialog.test.tsx

```diff
@@ -0,0 +1,157 @@
+    });
+    const assetASecondQueryMock = buildQueryMock<
+      PartitionHealthQuery,
+      PartitionHealthQueryVariables
+    >({
+      query: PARTITION_HEALTH_QUERY,
+      variables: {assetKey: {path: ['asset_a']}},
+      data: {assetNodeOrError: buildAsset('asset_a', ['test', 'test2'])},
+    });
+    const assetBSecondQueryMock = buildQueryMock<
+      PartitionHealthQuery,
+      PartitionHealthQueryVariables
+    >({
+      query: PARTITION_HEALTH_QUERY,
+      variables: {assetKey: {path: ['asset_b']}},
+      data: {assetNodeOrError: buildAsset('asset_b', ['test', 'test2'])},
+      delay: 5000,
+    });
+
+    const addPartitionMock = buildMutationMock<
+      AddDynamicPartitionMutation,
+      AddDynamicPartitionMutationVariables
+    >({
+      query: CREATE_PARTITION_MUTATION,
+      variables: {
+        repositorySelector: {repositoryName: 'test', repositoryLocationName: 'test'},
+        partitionsDefName: 'foo',
+        partitionKey: 'test2',
+      },
+      data: {
+        addDynamicPartition: buildAddDynamicPartitionSuccess(),
+      },
+    });
+
+    const assetAQueryMockResult = getMockResultFn(assetAQueryMock);
+    const assetBQueryMockResult = getMockResultFn(assetBQueryMock);
+    const assetASecondQueryMockResult = getMockResultFn(assetASecondQueryMock);
+    const assetBSecondQueryMockResult = getMockResultFn(assetBSecondQueryMock);
+    render(
+      <MemoryRouter>
+        <MockedProvider
+          mocks={[
+            assetAQueryMock,
+            assetBQueryMock,
+            assetASecondQueryMock,
+            assetBSecondQueryMock,
+            addPartitionMock,
+          ]}
+        >
+          <LaunchAssetChoosePartitionsDialog
+            open={true}
+            setOpen={(_open: boolean) => {}}
+            repoAddress={buildRepoAddress('test', 'test')}
+            target={{
+              jobName: '__ASSET_JOB_0',
+              partitionSetName: '__ASSET_JOB_0_partition_set',
+              type: 'job',
+            }}
+            assets={[assetA, assetB]}
+            upstreamAssetKeys={[]}
+          />
+        </MockedProvider>
+      </MemoryRouter>,
+    );
+
+    await waitFor(() => {
+      expect(assetAQueryMockResult).toHaveBeenCalled();
+      expect(assetBQueryMockResult).toHaveBeenCalled();
+    });
+
+    const link = await waitFor(() => screen.getByTestId('add-partition-link'));
+    userEvent.click(link);
+    const partitionInput = await waitFor(() => screen.getByTestId('partition-input'));
+    await userEvent.type(partitionInput, 'test2');
+    expect(assetASecondQueryMockResult).not.toHaveBeenCalled();
+    expect(assetBSecondQueryMockResult).not.toHaveBeenCalled();
+    const savePartitionButton = screen.getByTestId('save-partition-button');
+    userEvent.click(savePartitionButton);
+
+    await waitFor(() => {
+      expect(assetASecondQueryMockResult).toHaveBeenCalled();
+    });
+
+    expect(errorMock).not.toHaveBeenCalled();
```

Comment:
We can remove this expectation since we didn't use it, we relied on the error being thrown to fail the test as opposed to relying on an assertion failing to fail to the test

Reply:
got it, sounds good